src/main/java/fi/aalto/cs/apluscourses/intellij/services/PluginSettings.java

```diff
@@ -12,8 +15,14 @@
 
 public class PluginSettings implements MainViewModelProvider {
 
+  public interface LocalSettingsNames {
```

Comment:
IIRC, Sonar would prefer this to be a (static) class. For me, it doesn't matter.

Reply:
hm... but the report is clean (at least it does not complain about it, but some other things), and it's an **interface**, does it make any difference?

Reply:
I meant that "Move constants to a class or enum" code smell. But surely it's a minor issue.

Reply:
I dislike this `ENUM.BLAAAAH.name()` approach to get the value and also the amount of code for `enum` exceeds so much the same for the `interface` approach, that I'd prefer to keep it for now :pleading_face:

Reply:
Totally agree here.

I also kinda dislike enums in Java so let's forget them. What I was saying is that Sonar (for some imo-not-so-valid reason) likes it more if `LocalSettingsNames` is a class instead of interface (and it probably should have a private constructor too). Personally I see interface better here, just as you coded.